src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Window.cs

```diff
@@ -3059,6 +3092,35 @@ private static void OnBackdropTypeChanged(DependencyObject d, DependencyProperty
             WindowBackdropManager.SetBackdrop(window, (WindowBackdropType)e.NewValue);
         }
 
+        internal void UpdateBackdrop()
+        {
+            if (SystemParameters.HighContrast 
+                    || (!ThemeManager3.IsFluentThemeEnabled && this.Theme == "None"))
+            {
+                this.WindowBackdropType = WindowBackdropType.None;
+            }
+            else
+            {
+                this.WindowBackdropType = WindowBackdropType.MainWindow;
+            }
+        }
+
+        // Is this a good idea to move this method here ? I tried doing it to avoid reapplication
+        internal void SetImmersiveDarkMode(bool useDarkMode)
+        {
+            if(!Standard.Utility.IsOSWindows11OrNewer) return;
+
+            if(useDarkMode != _useDarkMode)
+            {
+                IntPtr handle = CriticalHandle;
+                if (handle != IntPtr.Zero)
+                {
+                    SNM.DwmSetWindowAttributeUseImmersiveDarkMode(handle, useDarkMode);
+                    _useDarkMode = useDarkMode;
```

Comment:
`DwmSetWindowAttributeUseImmersiveDarkMode` is not declared with `SetLastError = true`. In that case you should check the result and if it failed, not update your state.

Reply:
I was doing the check on value returned but got some errors as soon as I moved it to window class, I will reintroduce this.